pkg/asset/installconfig/azure/metadata.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+
+	typesazure "github.com/openshift/installer/pkg/types/azure"
+)
+
+// Metadata holds additional metadata for InstallConfig resources that
+// does not need to be user-supplied (e.g. because it can be retrieved
+// from external APIs).
+type Metadata struct {
+	session   *Session
+	client    *Client
+	dnsCfg    *DNSConfig
+	cloudName typesazure.CloudEnvironment
```

Comment:
So we need this information to re-generate a session. When install config is stored in the state it will not be stored in state and when we need the cloud after loading install config asset from state we will no longer have the cloudName configured.

Any information that is required to create a new metadata must be Serialized to json..